### ISSUE TITLE
GATED research: Score readability preference / reward reranking

### ISSUE THREAD

**Issue by gr-rr** [ListenCloser/listencloser#1308]
Program owner: #1303.
Research synthesis: #1297.
Score authority/program: #498 / #945.
Relevant current candidate work: MuseScore baseline, PM2S challenger, future Rubato-class systems when operationally usable.
Representation authority: #613.

## Status

**GATED SECOND WAVE — do not start until stable competing Score candidates and a small pairwise readability dataset exist.**

This issue formalizes the cheapest credible version of the “reward/preference learning for Score readability” idea. It is explicitly **not an RL issue**.

## Question

> **Can a small learned preference/reward model rerank valid Score candidates toward human-readable notation while preserving performance fidelity better than fixed handcrafted scoring?**

## Why this is well motivated

ListenCloser already distinguishes:

```text
canonical performance MIDI
≠
readable inferred Score
```

The Score task has multiple objectives:
- note/event preservation;
- timing/playback fidelity;
- metrical validity;
- readable notation;
- voice/staff continuity;
- notational economy.

Recent work such as Rubato and 2026 dual-evaluation research makes clear that notation similarity/readability and playback similarity are complementary, not interchangeable objectives.

This makes Score a natural candidate for **candidate generation + constrained/ranked selection** before any attempt at sequential RL editing.

## Gate to start

Do not train a preference model until:

- [ ] at least two legitimate candidate-generation routes can produce comparable Score outputs for the same source (e.g. MuseScore baseline + PM2S; later another candidate only if available);
- [ ] candidate outputs preserve exact source/provenance and can be rendered/parsed reproducibly;
- [ ] objective fidelity/validity features can be computed for each candidate;
- [ ] at least ~100 pairwise readability judgments exist across multiple pieces/passages, with a target of ~200–500 if collection is cheap;
- [ ] train/dev/test split is by piece/source, not by candidate pair;
- [ ] #498/#945 have not already settled the product decision in a way that makes this pilot unnecessary.

## Candidate pair unit

A judgment should compare two renderable Scores derived from the same canonical source evidence:

```text
source performance MIDI / exact Version
candidate A score + provenance
candidate B score + provenance
preferred for reading: A | B | tie/unclear
```

The human question should focus on readability/usability, not “which sounds more like the source?”; playback fidelity is separately measurable.

## Hard constraints / automatic features

Keep invalid candidates out before learning preference where possible:

- parse/render validity;
- catastrophic note loss/hallucination checks;
- source/event preservation;
- measure arithmetic validity;
- playback/timing similarity where applicable;
- exact Version/provenance lineage.

Candidate-level structured features may include:
- tie/rest/tuplet fragmentation;
- voice changes;
- staff changes/cross-staff behavior;
- rhythmic complexity;
- spelling/accidental stability;
- measure density;
- notational economy;
- objective notation-reference metrics when legitimate references exist;
- playback similarity metrics.

Do not use a single weighted heuristic score as “ground truth.”

## Models

Start small:

```text
A. deterministic handcrafted/objective ranking baseline
B. logistic / Bradley-Terry pairwise preference model
C. tiny MLP only if nonlinear interactions appear useful
```

A model predicts pairwise readability preference or a scalar score used only for reranking valid candidates.

No policy-gradient/GRPO/DPO/sequence-model fine-tuning in this issue.

## Metrics

Primary:
- held-out pairwise preference accuracy by source piece.

Secondary:
- rank correlation / NDCG when >2 candidates exist;
- calibration;
- agreement vs human inter-rater consistency;
- fidelity constraints on chosen candidate relative to baseline;
- correction burden proxy only if measured honestly.

A successful preference model must not improve readability by selecting candidates with materially worse source fidelity.

## Compute budget

Expected CPU-only for the reward/ranker itself.

```text
GPU-hours = 0 preferred
cash cost = $0 for training
```

Any GPU cost of candidate generation belongs to the existing candidate/runtime owners, not this research model.

## Success criterion

A strong result is:

> The tiny learned ranker predicts held-out readability preferences better than fixed objective/handcrafted ranking while satisfying predeclared fidelity/validity constraints.

This would justify a focused reranking/product experiment.

## Stop / kill criteria

Stop if:

- candidate systems rarely disagree meaningfully;
- human readability judgments are too inconsistent to learn;
- a small set of deterministic objective rules already reaches agreement ceiling;
- improved preference accuracy comes from fidelity degradation;
- the data requirement becomes hundreds/thousands of expert score annotations rather than cheap pairwise choices;
- stable Score candidate infrastructure is not ready.

## Later escalation — explicitly out of scope

Only after a reward/ranker succeeds should a separate research decision consider:
- search over discrete notation edits;
- learned edit policies;
- RL/post-training of a score generator;
- preference optimization.

Do **not** infer that RL is justified merely because this issue exists.

## Deliverables

1. Fixed source/candidate manifest with exact provenance.
2. Objective validity/fidelity feature extractor.
3. Pairwise readability dataset with source-disjoint split.
4. Deterministic vs learned small-ranker comparison.
5. Human agreement/noise analysis.
6. Fidelity guardrail analysis for preferred selections.
7. Explicit disposition:

```text
FAIL — preference not learnable/useful beyond deterministic scoring
NARROW — measurable but insufficient product lift
CONTINUE — held-out readability lift under fidelity constraints; open focused reranking owner
```

## Acceptance

- [ ] start gates satisfied;
- [ ] performance MIDI remains canonical and untouched;
- [ ] only valid/provenance-exact Score candidates compared;
- [ ] human readability and playback fidelity kept as separate axes;
- [ ] small CPU model vs deterministic baseline evaluated on held-out sources;
- [ ] no RL/generator fine-tuning introduced;
- [ ] disposition written back to #1303 and #498/#945 as appropriate.

Refs #1303 #1297 #498 #945 #613.

**Comment by gr-rr** [comment]
## Program constraint update: no bespoke manual labeling assumed

Current #1303 execution policy assumes no dedicated human readability-labeling program.

Keep this issue gated until stable same-source competing Score candidates **and** legitimate human/public pairwise readability evidence exist. Objective notation/fidelity metrics remain useful constraints/diagnostics but are not a substitute for the preference target.

Do not use an LLM/vision model as held-out readability ground truth. It may be used explicitly as weak supervision or to propose/error-cluster candidate features, provided evaluation remains against legitimate external evidence.

Until the data gate clears, compute cost is irrelevant and no model fitting should start. Refs #1303 #498 #945 #1311.
